### DISCUSSION TITLE
Both Brightness and Volume Control of DELL UP2715K not Working in Dual mDP of 5K Mode

### DISCUSSION THREAD

**Discussion by exsystem** [MonitorControl/MonitorControl#613] [General]
### Before opening the issue, have you...?

- [X] Searched for existing issues
- [X] I have read through [the Monitor-Troubleshooting Wiki](https://github.com/MonitorControl/MonitorControl/wiki/Monitor-Troubleshooting)
- [X] Updated MonitorControl to the latest version (if applicable)

### Describe the issue

Volume & Brightness controls are not working on DELL UP2715K when in 5K mode using 2 mDP ports connecting to Mac Pro 2013 (dual D700 model), however using single mDP port in 4K mode does working fine.

### Expected behavior

Expected for working on 5K mode with dual mDP ports.

### Anything else?

When testing, I have an additional DELL E2213H display connected via HDMI also connected, which works fine for brightness control(No Volume control needed since no built-in speakers for this display) with whatever mode the DELL UP2715K is in (single mDP with 4K or dual mDP with 5K).

Those 2 mDP to DP cables are in DP1.2 version as the seller told me, however I've no way to tested it out (Is there any detecting device? No display and audio issue for using those cables with my Mac Pro and display). 

### Environment Information (please complete the following information)

```markdown
- macOS version: 11.5.2
- Mac model: Mac Pro (Late 2013)
- MonitorControl version: 3.1.0 Build 3364 - Intel
- Monitor(s): DELL UP2715K and DELL E2213H
- Monitor Cable(s)/Connection(s): dual mDP to DP for DELL UP2715K, HDMI to DVI for DELL E2213H
- Apple Silicon/M1 (yes or no): no
```

**Comment by waydabber** [comment]
Hi. I see you are using an old 3.1.0 unsigned build. [Here is a much more recent one - v4.0.0 build 4527](https://www.dropbox.com/s/w1md5rjyll0lsyo/MonitorControl.zip?dl=1) (you have to enable Advanced preferences to see many of the new stuff).

This will probably have the same issue though.

Questions:

- What do you see on the Displays panel?
- Does software control work (if you force it by disabling DDC in the Displays panel) or nothing works at all?

Thanks!

**Comment by exsystem** [comment]
HI, 

For the v4.0.0 build 4527:
- The brightness control now works normal as expected, the brightness can be changed and the built-in macOS OSD can be shown. (Use hardware DDC control: checked)
- The volume control does not work, but I can see the built-in macOS OSD showing the value was changing as pressing volume control keys on my magic keyboard. The system volume slider inside the volume icon in menu bar is disabled ('grayed out'), as well as the Output volume slider is disabled in Sound Preference -> Output tab.

And I've noticed that I have installed a software called Boom3D (version 1.3.3), but I've turned it off and quit it, however I still see a process called "Core Audio Driver (BoomAudio.driver)" which I can't kill it. Is that matter?

Let me try uninstalling Boom3D first.But I wish the compatibility with it :)

**Comment by exsystem** [comment]
OK, Boom3D should be irrelevant. Since I forgot I can control the volume under 4K mode...

**Comment by waydabber** [comment]
Hi exsystem - are you sure the brightness control is hardware based? In 4.0.0 a combined control method is enabled by default (below a certain treshold software dimming kicks in as hardware brightness control is usually limited). If you don't experience any change over 50% then the hardware part still does not work. Thank you!

**Comment by exsystem** [comment]
> Hi exsystem - are you sure the brightness control is hardware based? In 4.0.0 a combined control method is enabled by default (below a certain treshold software dimming kicks in as hardware brightness control is usually limited). If you don't experience any change over 50% then the hardware part still does not work. Thank you!

OK, so it's still not working in hardware part. No brightness changes over 50%. Just not noticing the combined control method in 4.0.0.

**Comment by waydabber** [comment]
Can you try if other DDC based apps work? What happens if you try ddcctl or [Lunar](https://lunar.fyi)? I think the problem seems to be that because of the dual DP connection the system does not provide with the correct service port for DDC control. Version 4.0.0 now uses a private call that is provided by Apple to get the right port so the ambiguity of the former version in this regard should not be present anymore but it seems it still does not work.

You could also try if control works with lower resolution (try full HD or 4K) as in that case probably only one DP connection is utilized.

Your setup is rather unique (there are not many monitors like this around connected to older mac pros... :)) but as I see others had similar issues in the past https://github.com/MonitorControl/MonitorControl/issues/283.

@alin23

**Comment by sergiughiran** [comment]
I have a similar issue on a Dell S2721DGFA monitor connected to an M1 macbook pro. The connection is made via a DP cable plugged in a Caldigit soho dock (the dock then connects to the monitor). I tried the ddc alternative( m1ddc) but I get this: I2C communication failure: (iokit/?) unknown subsystem error.

**Comment by waydabber** [comment]
@sergiughiran - please try it without the Dock. Since the dock in question has two display outputs and some special internal logic to handle mirroring at 60Hz it probably does something weird with i2c communication which is needed for DDC.

**Comment by sergiughiran** [comment]
I tried it with a HDMI to USB-C cable and it seemed to work fine. I mention that I am using the macbook in clamshell mode with only one display connected to it. Is there no way to make it work through the dock? Thanks!

**Comment by waydabber** [comment]
No, probably this dock blocks DDC. You should inquire about this at the manufacturer, maybe they can help or there is a firmware update.

**Comment by waydabber** [comment]
Hi @exsystem, since there is no much to do with this at the moment, I'll move this over to discussion, maybe others with similar displays can add their experience. It would be great if you could try if control works with lower resolution (try full HD or 4K) as in that case probably only one DP connection is utilized.

**Comment by tooi96segment** [comment]
Hi,

Sorry to revive an old thread… did anyone ever manage to resolve this?
Looked through the different discussions and no apparent solution.
Using v 4.3.3.

To summarise, the issue is with the **dual** connection to achieve 5K. As long as only a **single** cable is connected all is fine. Except for losing 5k and being downgraded to 4k or less…
Connect the second cable and no longer able to use DDC comms.

**One cable**
```
ddcctl -d 1 -v \?
D: CGDisplay 6A173DC3-00E2-A536-805F-4A36711A1976 dispID(#722474197) (2560x1440 0°) 109.00 DPI
I: found 1 external display
E: Failed to find display in WindowServer's preferences! (/Library/Preferences/com.apple.windowserver.plist)
I: polling EDID for #1 (ID 722474197 => (null))
I: got edid.serial: 3D8F868A219S
I: got [edid.name](http://edid.name/): DELL UP2715K
D: action: v: ?
D: querying VCP control: #98 =?
I: VCP control #98 (0x62) = current: 50, max: 100
```

**Two cables**
```
ddcctl -d 1 -v \?
D: CGDisplay 788965F1-39AF-63EC-08F4-0A5F8E2487EE dispID(#722474196) (2560x1440 0°) HiDPI
I: found 1 external display
E: Failed to find display in WindowServer's preferences! (/Library/Preferences/com.apple.windowserver.plist)
I: polling EDID for #1 (ID 722474196 => (null))
I: got edid.serial: 3D8F868A219S
I: got [edid.name](http://edid.name/): DELL UP2715K
D: action: v: ?
D: querying VCP control: #98 =?
E: No data after 10 tries! (30000001ns reply-timeout)
E: DDC send command failed!
E: VCP control #98 (0x62) = current: 0, max: 0
```

It seems the specific issue here is with the software finding the (correct) path to the monitor while connected over two cables.
There is a primary and a secondary cable, at least for the Dell UP2715K. Connect them in the wrong order and you may loose POST/bootscreen, etc.

Would it be possible to fix this by somehow forcing all DDC communication over the primary cable? Telling the system which route to use? It might be ambiguous to the system currently.

Thanks,

Mark
